aiohasupervisor/models/os.py

```diff
@@ -0,0 +1,127 @@
+"""Models for OS APIs."""
+
+from dataclasses import dataclass
+from enum import StrEnum
+from pathlib import PurePath
+
+from .base import Options, Request, ResponseData
+
+# --- ENUMS ----
+
+
+class RaucState(StrEnum):
+    """RaucState type."""
+
+    GOOD = "good"
+    BAD = "bad"
+    ACTIVE = "active"
+
+
+class BootSlotName(StrEnum):
+    """BootSlotName type."""
+
+    A = "A"
+    B = "B"
+
+
+# --- OBJECTS ----
+
+
+@dataclass(frozen=True, slots=True)
+class BootSlot(ResponseData):
+    """BootSlot model."""
+
+    state: str
+    status: RaucState | None
+    version: str | None
+
+
+@dataclass(frozen=True, slots=True)
+class OSInfo(ResponseData):
+    """OSInfo model."""
+
+    version: str | None
+    version_latest: str | None
+    update_available: bool
+    board: str | None
+    boot: str | None
+    data_disk: str | None
+    boot_slots: dict[str, BootSlot]
+
+
+@dataclass(frozen=True, slots=True)
+class OSUpdate(Request):
+    """OSUpdate model."""
+
+    version: str | None = None
+
+
+@dataclass(frozen=True, slots=True)
+class MigrateDataOptions(Request):
+    """MigrateDataOptions model."""
+
+    device: str
+
+
+@dataclass(frozen=True, slots=True)
+class DataDisk(ResponseData):
+    """DataDisk model."""
+
+    name: str
+    vendor: str
+    model: str
+    serial: str
+    size: int
+    id: str
+    dev_path: PurePath
+
+
+@dataclass(frozen=True, slots=True)
+class DataDiskList(ResponseData):
+    """ListDataDisks model."""
+
+    devices: list[str]
```

Comment:
Good catch! This property wasn't marked as deprecated in the API but looking at it I think it should be. This API used to return only a list of device IDs and nothing else. `disks` was added in [this PR](https://github.com/home-assistant/supervisor/pull/4202) to give users much more information about the available disks and it includes the ids as well.

I don't think we can remove it from Supervisor's API yet because the existing change datadisk feature likely relies on it. But no reason to add it to the client library.